alpha/lib/model/asset.php

```diff
@@ -545,7 +545,10 @@ public function getDownloadUrl($useCdn = false, $forceProxy = false, $preview =
 		$syncKey = $this->getSyncKey(self::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
 		list($fileSync, $serveRemote) = kFileSyncUtils::getFileSyncByStoragePriority($this->getPartnerId(), $syncKey, true);
 		
-		if($serveRemote && $fileSync && !in_array($fileSync->getDc(), kDataCenterMgr::getSharedStorageProfileIds()))
+		//Shared storage file syncs should not be handled as remote
+		$serveRemote = $serveRemote && !in_array($fileSync->getDc(), kDataCenterMgr::getSharedStorageProfileIds());
```

Comment:
i think the original condition was more correct as you access $fileSync->getDc but here only in line 551 you check if the object is not null instead of having it before

Reply:
Done.